Borg Module Rework

whatston3 opened the pull request:
<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR
<!-- What did you change? -->
<!-- If this is a code change, summarize at high level how your new code works. This makes it easier to review. -->

Adds the ability for borg modules to start with empty placeholder items instead of proper hands.

Reviews a bunch of cyborg modules.  In general, more receive "hands".
- Engineering cyborgs start with a shipyard RCD with a hand for ammo instead of an infinite ammo, gridless RCD.
- The experimental RCD module has been moved to a T3 research tech.
- The construction module comes with a construction bag, as well as a set of generic hands for construction materials and one for floor tiles.
- The service module has been deprecated and separated into two modules, one for mixing and serving drinks and smokes, and one for bureaucracy.  The service borg starts with both of these.
- All cyborg chassis receive a jetpack module on spawn - this jetpack module contains a mass scanner and GPS for navigation.  We're the big open fork, we can embrace this.  [Added a cash hand and a clock, because why not?]
- The cleaning module's spray bottle has been made droppable.
- The light replacer module receives a hand to hold lightbulbs.
- The appraisal module now has two hands for trading crates.  I think the borgs should be able to interact with the racks, but this is left as a TODO for now.
- Adds a chef module that includes a non-droppable kitchen knife, rolling pin, a slot for a beaker and two generic "ingredient slots".
- The treatment module has been made to work with arbitrary healing items (thank you, @deltanedas).  The syringe is also droppable and replaceable with a bluespace version.
- The advanced paramedic module has been repurposed as a stand-in for medibots - you have infinitely recharging hypos filled with tricordrazine and inaprovaline.
- The clowning module receives a slot for banana peels and banana cream pies.

~~Grabs everything from #3054 (thank you @EckoAurum) - this can be merged before or after.~~ already merged

Cherry-picks features from:
[Delta-V#3119 (borg modules)](https://github.com/DeltaV-Station/Delta-v/pull/3119)
[Delta-V#3168 (experimental RCD rework)](https://github.com/DeltaV-Station/Delta-v/pull/3168)
And probably bugfixes from Delta-V#3131, Delta-V#3132, and Delta-V#3155.

I owe @deltanedas a liver, kidney or lung for this one.

## Why / Balance
<!-- Discuss how this would affect game balance or explain why it was changed. Link any relevant discussions or issues. -->

Borg quality of life.  Engineering borg shouldn't start at the absolute apex of its existence, let it yearn.

## Technical details
<!-- Summary of code changes for easier review. -->

Added a placeholders field to the DroppableBorgModuleComponent.
Items here specify their display item's ID and their expected item whitelist - they do not start with an item and must spawn a display standin.
Item insertion and removal now has to happen over both the droppable item set and the placeholder set.

## How to test
<!-- Describe a procedure to test this feature, along with expected output/behavior. -->

Play around with cyborgs, try to use each of the module's items, try dropping and reinserting droppable items, try to find odd items that go in droppable slots (the ingredient one will be scuffed, not much we can do there)

## Media
<!-- Attach media if the PR makes ingame changes (clothing, items, features, etc). 
Small fixes/refactors are exempt. Media may be used in SS14 progress reports with credit. -->

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [X] I have read [CONTRIBUTING.md](https://github.com/new-frontiers-14/frontier-station-14/blob/master/CONTRIBUTING.md) and and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.
- [X] I have reviewed the [Ship Submission Guidelines](https://frontierstation.wiki.gg/wiki/Ship_Submission_Guidelines) if relevant.
- [X] I confirm that AI tools were not used in generating the material in this PR.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them. -->

**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->
<!--
:cl:
- add: Added fun!
- remove: Removed fun!
- tweak: Changed fun!
- fix: Fixed fun!
-->
:cl: whatston3 and deltanedas
- tweak: All cyborgs now start with a jetpack module with built-in cash hand and clock. Engineer borgs start with a shipyard RCD module.
- tweak: More cyborg modules have slots for placeholder items.
- add: A new module for chef cyborgs can be crafted with research.

deltanedas commented:
dibs on the kidney :trollface:

whatston3 commented:
Made a handful of changes.  Need to do some more testing, but I think this _should_ be about alright.

The jetpack module that the cyborgs start with now has a hand for cash, another for fuel and an undroppable clock.  This should give some decent quality of life and help in general autonomy and keeping a shuttle running.

One thing that's probably a bit jank is that I've allowed borgs to bypass the item component check depending on a flag in the hand placeholder, which is set on requested initial items (or placeholder items) that do not have an ItemComponent.  With this, we can allow salvage borgs to haul trading crates.

EckoAurum commented:
Amazing work!! Besides the E-dagger which has been fixed, and the Chemical Recovery module Borg Hypo parenting issue: It's a bit strange that the engineering board cannot pickup computer boards, unsure if this is intentional or not. The Gardening Module does not have a whitelist for Seeds, they can only harvest and collect produce but cannot plant. The appraisal module, you can REALLY throw theses crates with some force and easily send people into walls, May not be a massive issue, but I can certainly foresee borgs playing "physics dodgeball"

whatston3 commented:
Seeds: fair, added a dropper as well since I think it was removed from the NFService module.
Computer/machine boards: eh, sure, but it's a little bit of a moot point since constructing machines potentially requires a bunch of weird objects.  Flatpacks though, sure.
Borg hypo: fixed, added fill states.
![image](https://github.com/user-attachments/assets/1fb84385-5618-4dbb-9f1e-49d41047f3a4)

whatston3 commented:
Added computer boards to the engineering cable module.
Added a flatpack to the engineering construction module.
Added more descriptive, localized description strings for item slots and placeholders.

ErhardSteinhauer reviewed (approved):
Fiddled about with it for a bit; it seems to be working fine.
